config/module/tree.go

```diff
@@ -92,6 +92,25 @@ func (t *Tree) Children() map[string]*Tree {
 	return t.children
 }
 
+// DeepEach calls the provided callback for the receiver and then all of
+// its descendents in the tree, allowing an operation to be performed on
+// all modules in the tree.
+//
+// Parents will be visited before their children but otherwise the order is
+// not defined.
+func (t *Tree) DeepEach(cb func(*Tree)) {
```

Comment:
Yes, this is used by the code that will be in the next PR to visit all of the loaded modules and get their version constraints. Honestly I ought to have kept this back for the next PR but I guess I just missed when I was somewhat-arbitrarily choosing a point in history to submit as a PR here.